mpd 0.22.6 doesn't build in Darwin

**Describe the bug**
mpd 0.22.6 doesn't build in Darwin, because it requires C++17.

**To Reproduce**
Log from latest Hydra job:

https://hydra.nixos.org/build/139209751/nixlog/1

Relevant lines:
```
../src/ls.cxx:46:14: error: no member named 'merge' in 'std::__1::set<std::__1::basic_string<char>, std::__1::less<std::__1::basic_string<char> >, std::__1::allocator<std::__1::basic_string<char> > >'
                        protocols.merge(plugin.protocols());
                        ~~~~~~~~~ ^
../src/ls.cxx:44:29: warning: lambda capture 'protocols' is not required to be captured for this use [-Wunused-lambda-capture]
        decoder_plugins_for_each([&protocols](const auto &plugin){
                                  ~^~~~~~~~~
../src/ls.cxx:66:14: error: no member named 'merge' in 'std::__1::set<std::__1::basic_string<char>, std::__1::less<std::__1::basic_string<char> >, std::__1::allocator<std::__1::basic_string<char> > >'
                        protocols.merge(plugin.protocols());
                        ~~~~~~~~~ ^
../src/ls.cxx:64:37: warning: lambda capture 'protocols' is not required to be captured for this use [-Wunused-lambda-capture]
        decoder_plugins_for_each_enabled([&protocols](const auto &plugin){
                                          ~^~~~~~~~~
2 warnings and 2 errors generated.
```

**Additional context**
The error is related to `std::set::merge` not found. It was added in C++17.

**Notify maintainers**

@tobim @fpletz 

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: [mpd, mpd-small]
```

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -18316,7 +18316,7 @@ in
   mod_pkcs12 = pkgs.apacheHttpdPackages.mod_pkcs12;
   mod_timestamp = pkgs.apacheHttpdPackages.mod_timestamp;
 
-  inherit (callPackages ../servers/mpd { })
+  inherit (callPackages ../servers/mpd { stdenv = if stdenv.cc.isClang then llvmPackages_8.stdenv else stdenv; })
     mpd mpd-small mpdWithFeatures;
 
   libmpdclient = callPackage ../servers/mpd/libmpdclient.nix { };

